src/utils/api/returnValues/shoppingLists.d.ts

```diff
@@ -2,6 +2,55 @@ import { ApiResponse, HTTPHeaders } from '../http'
 import { ErrorObject, ResponseShoppingList } from '../../../types/apiData'
 import { UnauthorizedResponse } from './shared'
 
+/**
+ *
+ * Types used for POST /games/:game_id/shopping_lists endpoint
+ *
+ */
+
+class PostShoppingListsSuccessResponse extends ApiResponse {
+  status: 201
+
+  constructor(
+    body,
+    options: { status: 201; statusText?: string; headers?: HTTPHeaders }
+  ) {
+    super(body, options)
+  }
+}
+
+class PostShoppingListsNotFoundResponse extends ApiResponse {
+  status: 404
+
+  constructor(
+    body,
+    options: { status: 404; statusText?: string; headers?: HTTPHeaders }
+  ) {
+    super(body, options)
+  }
+}
+
+class PostShoppingListsErrorResponse extends ApiResponse {
+  status: 422 | 500
+
+  constructor(
+    body,
+    options: { status: 422 | 500; statusText?: string; headers?: HTTPHeaders }
+  ) {
+    super(body, options)
+  }
+}
+
+export type PostShoppingListsResponse =
+  | UnauthorizedResponse
+  | PostShoppingListsSuccessResponse
+  | PostShoppingListsNotFoundResponse
+  | PostShoppingListsErrorResponse
+
+export type PostShoppingListsReturnValue =
+  | { status: 201; json: ResponseShoppingList[] }
+  | { status: 422 | 500; json: ErrorObject }
```

Comment:
This is not actually a possible return value so I'm not sure how I feel about it. In the error case an `ApiError` would be thrown. I might try removing it and see if it makes anything break or, alternatively, better.

Reply:
Update: Nope, I tried a couple different approaches to change this but TypeScript isn't smart enough to connect the different types of `PostShoppingListsResponse` object to the different types of `PostShoppingListsReturnValue`. The more I think about it, the more it seems like the assumption that these types do, in fact, correspond is not type safe, in light of the fact API responses could change.